common/overlaysynccleanuphandler.h

```diff
@@ -43,6 +43,17 @@ class OverlaySyncCleanupHandler : public CleanUpHandler
      * @return ReturnCode::FAILURE on failure
      */
     ReturnCode clean() override;
+
+    /**
+     * @brief queryName is needed to query member name yet its full functionality is not needed for
+     * this class.
+     *
+     * Since OverlaySyncCleanupHandler uses DirectoryCleanUpHandler for removing directory, this
+     * function won't be called for the same reason.
+     *
+     * @return an empty string
+     */
+    const std::string queryName() override;
```

Comment:
Sure, but this check is done for anything we want to add a cleanup handler for, so it is a little more generic than just directories.